addons/web/static/src/core/ui/ui_service.js

```diff
@@ -31,13 +33,42 @@ export function useActiveElement(refName) {
         (el) => {
             if (el) {
                 uiService.activateElement(el);
+                trapFocus(el);
                 return () => uiService.deactivateElement(el);
             }
         },
         () => [owner.el]
     );
 }
 
+function trapFocus(element) {
+    const focusableEls = getTabableElements(element);
+    const firstFocusableEl = focusableEls[0];
+    const lastFocusableEl = focusableEls[focusableEls.length - 1];
+
+    element.addEventListener("keydown", function (e) {
+        const keyPressed = getActiveHotkey(e);
+        switch (keyPressed) {
+            case "tab":
+                if (document.activeElement === lastFocusableEl) {
+                    firstFocusableEl.focus();
+                    e.preventDefault();
+                }
+                break;
+            case "shift+tab":
+                if (document.activeElement === firstFocusableEl) {
+                    lastFocusableEl.focus();
+                    e.preventDefault();
+                }
+                break;
+        }
+    });
+
+    if (!(document.activeElement in element) && firstFocusableEl) {
+        firstFocusableEl.focus();
+    }
```

Comment:
even if it looks hacky/fragile(?), maybe try the following
```suggestion
    if (document.activeElement in element) {
        return;
    }
    if(firstFocusableEl)
        firstFocusableEl.focus();
    } else {
        const isItselfTabable = getTabableElements(element.parentElement).includes(element);
        if(isItselfTabable) {
            element.focus();
        } else {
            element.tabIndex = -1;
            element.focus();
        }
    }
```